rclpy/src/rclpy/_rclpy_pycapsule.c

```diff
@@ -0,0 +1,161 @@
+// Copyright 2019 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <Python.h>
+
+/// Get the name of a pycapsule.
+/**
+ * Raises TypeError if the argument is not a pycapsule
+ *
+ * \param[in] pycapsule a pycapsule
+ * \return Name or None if the capsule has no name.
+ */
+static PyObject *
+rclpy_pycapsule_name(PyObject * Py_UNUSED(self), PyObject * args)
+{
+  PyObject * pycapsule;
+  if (!PyArg_ParseTuple(args, "O", &pycapsule)) {
+    return NULL;
+  }
+
+  const char * name = PyCapsule_GetName(pycapsule);
+
+  if (PyErr_Occurred()) {
+    return NULL;
+  }
+
+  if (NULL == name) {
+    Py_RETURN_NONE;
+  }
+  return PyUnicode_FromString(name);
```

Comment:
I assume returning `NULL` here is the right thing to do if `FromString` fails?

Reply:
Correct,  [`PyUnicode_FromString`](https://docs.python.org/3/c-api/unicode.html#c.PyUnicode_FromString) returns `NULL` if it sets an exception ([second paragraph](https://docs.python.org/3/c-api/intro.html#exceptions)), and this function is required to do the same: https://docs.python.org/3/c-api/structures.html#c.PyCFunction